frontend/src/Menu/index.jsx

```diff
@@ -9,7 +11,8 @@ import leaderBoardMenuIcon from '../../assets/menu/Leaderboard.png';
 const HEIGHT_RATIO = 0.25;
 
 function Menu() {
-  const [isPressed, setPressed] = useState('Buildings');
+  const activatingComponent = useSelector((state) => state.menu.currentlyDisplayComponent);
```

Comment:
activating is an action, but it's referring to a component which is currently displayed
so it should be activeComponent or something like that
activating would be a good name if it was an action which activates something that was disabled before
from this name I would assume it's a function, but it's not, so the name should be changed